App/ApiRest/Routers.go

```
@@ -0,0 +1,134 @@
+package ApiRest
+
+import (
+	"net/http"
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"github.com/kjk/betterguid"
+	"github.com/yoiner-castillo-globant/GBcamp/App/ApiRest/Control"
+	"github.com/yoiner-castillo-globant/GBcamp/App/Request"
+)
+
+var IControl = Control.New()
+
+func NewCartEP(w http.ResponseWriter, req *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	newId := betterguid.New()
+	IControl.AddCart(newId)
+
+	w.WriteHeader(http.StatusCreated)
+	w.Write([]byte(`{"CartId": "` + newId + `"}`))
+}
+
+func GetItemsCartEP(w http.ResponseWriter, req *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	pathParams := mux.Vars(req)
+	CartId := pathParams["CartId"]
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(IControl.GetItems(CartId))
+}
+
+func validateAddItemRequest(w http.ResponseWriter, cartId, articleId string) bool {
+	if cartId == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(`{"Response": "you need to send an CartId"}`))
+		return false
+	}
+
+	if articleId == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(`{"Response": "you need to send an id"}`))
+		return false
+	}
+
+	return true
+}
+func AddItemCartEP(w http.ResponseWriter, req *http.Request) {
+	params := mux.Vars(req)
+	var article Request.PostArticle
+	cartId := params["CartId"]
+
+	_ = json.NewDecoder(req.Body).Decode(&article)
```

ignoring errors can be dangerous. If nothing can be returned it is a good idea to log a message